### PR TITLE
Add ExcludeFromCodeCoverageAttribute

### LINKED ISSUE #10685: [ExcludeFromCodeCoverage] is marked as internal
The ExcludeFromCodeCoverage attribute is marked as internal.  This is more than likely intended (It's also internal in the PCL) but it's public in all other versions of the .Net framework.

This attribute is as useful for user code as I'm sure it is in the framework's codebase.  Is there a particular design reason this was marked as internal?  Is there an alternative I'm not aware of?  If not, can it be made public?

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes #10685

@ericstj is there any meta file that I missed editing?

### CODE REVIEW

**Comment by danmosemsft** [comment]
@wesh look ok?

File: src/System.Diagnostics.Tools/tests/System/Diagnostics/CodeAnalysis/ExcludeFromCodeCoverageAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using Xunit;
+
+namespace System.Diagnostics.CodeAnalysis.Tests
+{
+    public class ExcludeFromCodeCoverageAttributeTests
+    {
+        [Fact]
+        public void ExcludeFromCodeCoverageAttribute_Instantiate()
+        {
+            new ExcludeFromCodeCoverageAttribute();throw null;
```
Comment (stephentoub) [inline src/System.Diagnostics.Tools/tests/System/Diagnostics/CodeAnalysis/ExcludeFromCodeCoverageAttributeTests.cs]: How is it that all of the tests passed when there's this explicit ```throw null``` here?

Comment (danmosemsft) [inline src/System.Diagnostics.Tools/tests/System/Diagnostics/CodeAnalysis/ExcludeFromCodeCoverageAttributeTests.cs]: Uh - my bad, I was verifying that the tests were running. Clearly, they're not.

Comment (danmosemsft) [inline src/System.Diagnostics.Tools/tests/System/Diagnostics/CodeAnalysis/ExcludeFromCodeCoverageAttributeTests.cs]: It's not building the test .csproj with   '$(TargetGroup)'=='netstandard1.7'. Hmm.

**Comment by danmosemsft** [comment]
Actually, this may need to be a case of exposing from coreclr instead.

**Comment by danmosemsft** [comment]
@ericstj Can you confirm it's correct to put this into corefx ? I thought so as it's not exposed publicly from the base library, but I seem to remember you may have said otherwise.

**Comment by stephentoub** [comment]
@danmosemsft, what are the next steps on this one?  Are you still waiting for feedback from @ericstj?  Thanks.

**Comment by ericstj** [comment]
LGTM after rebase.

**Comment by danmosemsft** [comment]
Hit a CI timeout https://github.com/dotnet/corefx/issues/12252 but need to rebase anyway.

File: src/System.Diagnostics.Tools/tests/project.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "Microsoft.DotNet.BuildTools.TestSuite": "1.0.0-prerelease-00807-03"
   },
   "frameworks": {
-    "netstandard1.3": {}
+    "netstandard1.3": {},
+    "netstandard1.7": {}
   },
   "supports": {
     "coreFx.Test.netcore50": {},
```
Comment (weshaggard) [inline src/System.Diagnostics.Tools/tests/project.json]: you will need to add corefx.Test.netcoreapp1.1 to get your netstandard17 test configuration to run.

File: src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.builds [hunk 3]
```diff
@@ -2,11 +2,15 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="System.Diagnostics.Tools.Tests.csproj"/>
+  <Project Include="System.Diagnostics.Tools.Tests.csproj"/>
     <Project Include="System.Diagnostics.Tools.Tests.csproj">
       <TestTFMs>netcore50;net46</TestTFMs>
       <OSGroup>Windows_NT</OSGroup>
     </Project>
+    <Project Include="System.Diagnostics.Tools.Tests.csproj">
```
Comment (weshaggard) [inline src/System.Diagnostics.Tools/tests/System.Diagnostics.Tools.Tests.builds]: @joperezr  you are going to need to pick-up this change with your PR https://github.com/dotnet/corefx/pull/12241.

**Pull request merged**